workflow-testing/src/main/java/com/squareup/workflow/testing/RenderTester.kt

```diff
@@ -350,6 +433,58 @@ interface RenderTester<PropsT, StateT, OutputT, RenderingT> {
    * handle a workflow or worker output or a rendering event.
    */
   fun render(block: (rendering: RenderingT) -> Unit = {}): RenderTestResult<PropsT, StateT, OutputT>
+
+  /**
+   * Describes a call to
+   * [RenderContext.renderChild][com.squareup.workflow.RenderContext.renderChild].
+   *
+   * @param workflow The child workflow that is being rendered.
+   * @param props The props value passed to `renderChild`.
+   * @param outputType The [KType] of the workflow's output type.
+   * @param renderingType The [KType] of the workflow's rendering type.
+   * @param renderKey The string key passed to `renderChild`.
+   */
+  class RenderChildInvocation(
+    val workflow: Workflow<*, *, *>,
+    val props: Any?,
+    val outputType: KTypeProjection,
+    val renderingType: KTypeProjection,
+    val renderKey: String
+  )
+
+  sealed class ChildWorkflowMatch {
+    /**
+     * Indicates that the child workflow did not match the predicate and must match a different
+     * expectation. The test will fail if all expectations return this value.
+     */
+    object NotMatched : ChildWorkflowMatch()
+
+    /**
+     * Indicates that the workflow matches the predicate.
+     *
+     * @param childRendering The value to return as the child's rendering.
+     * @param output If non-null, [WorkflowOutput.value] will be "emitted" when this workflow is
+     * rendered. The [WorkflowAction] used to handle this output can be verified using methods on
+     * [RenderTestResult].
+     */
+    class Matched(
+      val childRendering: Any?,
+      val output: WorkflowOutput<Any?>? = null
+    ) : ChildWorkflowMatch()
+  }
+
+  /**
+   * Result of an [expectSideEffect] matcher function.
+   */
+  enum class SideEffectMatch {
+    /** Indicates that the side effect does not match the expectation. */
+    NOT_MATCHED,
+
+    /**
+     * Indicates that the side effect matches the expectation, but may also match other expectations.
+     */
+    MATCHED
+  }
```

Comment:
This enum used to have three values, but now that it's back down to two, it could probably be replaced by an enum without losing much expressiveness.